### PR TITLE
nogo: provide usage doc

### PR DESCRIPTION
**Pull request by sluongng**
**What type of PR is this?**

> Uncomment one line below and remove others.
>
> Bug fix
> Feature
Documentation
> Other

**What does this PR do? Why is it needed?**

Explain basic usage of nogo to new user.

Also provide a quick comparison between `nogo` and other popular linters
such as golangci-lint

**Other notes for review**

### CODE REVIEW

File: go/nogo.rst [hunk 1]
```diff
@@ -117,6 +121,54 @@ the ``TOOLS_NOGO`` list of dependencies.
...
+
+Because ``nogo`` benefits from Bazel's incremental build and caching, it is more suitable for
+large code bases. If you have a smaller code base, you could use ``golangci-lint`` instead.
+
+If ``golangci-lint`` takes a really long time to run in your repository, you could try to use
+``nogo`` instead.
+
+As of the moment of this writing, there is no way for ``nogo`` to apply the fixers coupled
+with the analyzers. So separate linters such as ``golangci-lint`` or ``staticcheck`` are more
+ergonomic to apply the fixes into the code base.
```
Comment (fmeum) [inline go/nogo.rst:170]:
```suggestion
ergonomic to apply the fixes to the code base.
```

**Review by fmeum — APPROVED** [review]

**Comment by sluongng** [comment]
shall we merge this PR? @fmeum @linzhp

File: go/nogo.rst [hunk 2]
```diff
@@ -117,6 +121,54 @@ the ``TOOLS_NOGO`` list of dependencies.
         visibility = ["//visibility:public"],
     )
 
+Usage
+---------------------------------
+
+``nogo`` is invoked automatically when building any Go target in your workspace.
```
Comment (fmeum) [inline go/nogo.rst:127]: But it does require setup, right? Should we mention this here?

Comment (sluongng) [reply go/nogo.rst:127]: How about 

> ``nogo``, when configured, is invoked automatically when building any Go target in your workspace.

Comment (sluongng) [reply go/nogo.rst]: Fixed

**Pull request merged**